Essentials/src/com/earth2me/essentials/commands/Commandtpaccept.java

```diff
@@ -45,6 +48,12 @@ public void run(final Server server, final User user, final String commandLabel,
             user.requestTeleport(null, false);
             throw new Exception(tl("requestTimedOut"));
         }
+        final TeleportAcceptEvent event = new TeleportAcceptEvent(requester, user, TeleportCause.COMMAND);
+        Bukkit.getServer().getPluginManager().callEvent(event);
+        if (event.isCancelled()) {
+            user.requestTeleport(null, false);
```

Comment:
Not sure how I feel about canceling also clearing the request, want further input

Reply:
Cancelling shouldn't clear the request - IMO plugins should do that manually if they need to.

Reply:
I didn't originally but that allows them spam the tpa command if it gets cancelled

Reply:
I could add a setter that allows you to make it consumable `TeleportAcceptEvent#setConsumeRequest()`